### DISCUSSION TITLE
Rate limiting for the Composer v2 metadata API

### DISCUSSION THREAD

**Discussion by nareshdevasani** [composer/composer#11335] [General]
Hello dear members,
I wanted to use below API to fetch metadata for multiple PHP packages.
GET https://repo.packagist.org/p2/[vendor]/[package].json

But at times I get below error:
dial tcp 139.99.120.115:443: connect: connection refused

Does that mean, I cannot fetch metadata for many(in my case 500 packages) packages?

**Comment by Seldaek** [comment]
You should be able to, but make sure you use http2 and a persistent http connection so that you don't create a new connection per file you download (which may trip up anti-DDoS mechanisms).
